libs/core/algorithms/tests/regressions/CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ set(tests
     static_chunker_2282
     transform_inclusive_scan_4786
     transform_inclusive_scan_4787
+    reduce_test
```

Comment:
I'd suggesting naming the test something like `reduce_6647` to disambiguate the name from any other existing test (on some platforms all test names have to be unique. Also, please maintain alphabetical ordering in this list here (nothing enforceable, however it helps maintaining tests).